drivers/modem/modem_cellular.c

```diff
@@ -1263,6 +1254,100 @@ static void modem_cellular_cmux_handler(struct modem_cmux *cmux, enum modem_cmux
 	}
 }
 
+MODEM_CHAT_SCRIPT_CMDS_DEFINE(get_signal_csq_chat_script_cmds,
+			      MODEM_CHAT_SCRIPT_CMD_RESP("AT+CSQ", csq_match),
+			      MODEM_CHAT_SCRIPT_CMD_RESP("", ok_match));
+
+MODEM_CHAT_SCRIPT_DEFINE(get_signal_csq_chat_script, get_signal_csq_chat_script_cmds,
+			 abort_matches, modem_cellular_chat_callback_handler, 2);
+
+static int modem_cellular_get_signal(const struct device *dev,
+				     const enum cellular_signal_type type,
+				     int16_t *value)
+{
+	int ret = -ENOTSUP;
+	struct modem_cellular_data *data = (struct modem_cellular_data *)dev->data;
+
+	if ((data->state != MODEM_CELLULAR_STATE_AWAIT_REGISTERED) &&
+	    (data->state != MODEM_CELLULAR_STATE_CARRIER_ON)) {
+		return -ENODATA;
+	}
+
+	/* Run chat script */
+	switch (type) {
+	case CELLULAR_SIGNAL_RSSI:
+		ret = modem_chat_run_script(&data->chat, &get_signal_csq_chat_script);
+		break;
+
+	case CELLULAR_SIGNAL_RSRP:
+	case CELLULAR_SIGNAL_RSRQ:
+		/* TODO: Run CESQ script */
+		ret = -ENOTSUP;
+		break;
+	}
+
+	/* Verify chat script ran successfully */
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Validate and set received value */
+	switch (type) {
+	case CELLULAR_SIGNAL_RSSI:
+		if (data->rssi == INT16_MAX) {
+			ret = -ENODATA;
+			break;
+		}
+
+		*value = data->rssi;
+		break;
+
+	case CELLULAR_SIGNAL_RSRP:
+	case CELLULAR_SIGNAL_RSRQ:
+		/* TODO: Validate and set values */
+		ret = -ENODATA;
+		break;
+	}
```

Comment:
I had this in my PR, but since you're messing around here anyways. I think all switch-case should have default statements since `type` is equivalent to an int.

Reply:
Added defaults to both switches :)